ricecooker/classes/nodes.py

```diff
@@ -798,3 +798,84 @@ def truncate_fields(self):
             q.truncate_fields()
 
         super(ExerciseNode, self).truncate_fields()
+
+class SlideshowNode(ContentNode):
+    """ Model representing Slideshows
+
+        Slideshows are sequences of "Slides", a combination of an image and caption.
+        Slides are shown in a specified sequential order.
+
+        Attributes:
+            source_id (str): content's original id
+            title (str): content's title
+            license (str or <License>): content's license
+            author (str): who created the content (optional)
+            aggregator (str): website or org hosting the content collection but not necessarily the creator or copyright holder (optional)
+            provider (str): organization that commissioned or is distributing the content (optional)
+            description (str): description of content (optional)
+            files ([<SlideImageFile>]): images associated with slides
+            thumbnail (str): local path or url to thumbnail image (optional)
+            extra_fields (dict): any additional data needed for node (optional)
+            domain_ns (str): who is providing the content (e.g. learningequality.org) (optional)
+    """
+    kind = content_kinds.SLIDESHOW
+
+    def __init__(self, source_id, title, license, slideshow_data=None, **kwargs):
+        if slideshow_data:
+            extra_fields = {'slideshow_data': slideshow_data}
+        else:
+            extra_fields = {'slideshow_data':[]}
+        # THe Node base class' __init__ method has:
+        #       for f in files or []:
+        #           self.add_file(f)
+        super(SlideshowNode, self).__init__(source_id, title, license, extra_fields=extra_fields, **kwargs)
+
+    def add_file(self, file_to_add):
+        """
+        Add a the SlideImageFile to the node's files and append slideshow metadata
+        to extra_fields['slideshow_data'] (list).
+        Args: file (SlideshowNode or ThumbnailFile): file model to add to node
+        Returns: None
+        """
+        from .files import File, SlideImageFile
+        assert isinstance(file_to_add, File), "Files being added must be instances of a subclass of File class"
+        # TODO asserr SlideshowNode OR ThumbnailFile
```

Comment:
This would be good to add here to make sure only `SlideshowFile` and `ThumbnailFile` can be added to `SlideshowNode`s